util/request_source.go

```diff
@@ -73,13 +94,32 @@ func IsRequestSourceInternal(reqSrc *RequestSource) bool {
 func (r *RequestSource) GetRequestSource() string {
 	// if r.RequestSourceType is not set, it's mostly possible that r.RequestSourceInternal is not set
 	// to avoid internal requests be marked as external(default value), return unknown source here.
-	if r == nil || r.RequestSourceType == "" {
+	if r == nil || (len(r.RequestSourceType) == 0 && len(r.ExplicitRequestSourceType) == 0) {
 		return SourceUnknown
 	}
+	clearEmpty := func(list []string) []string {
+		i := 0
+		for _, v := range list {
+			if len(v) != 0 {
+				list[i] = v
+				i++
+			}
+		}
+		return list[:i]
+	}
+	explicitSourceType := r.ExplicitRequestSourceType
+	if len(explicitSourceType) == 0 {
+		explicitSourceType = ExplicitTypeDB
```

Comment:
NIT: not sure if `db` is a good name for the default value. Maybe `default` is more conventional?